src/dataExplorer/context/fields.tsx

```diff
@@ -61,7 +62,11 @@ export const FieldsProvider: FC<Prop> = ({children, scope}) => {
     ? EXTENDED_TAG_LIMIT
     : DEFAULT_TAG_LIMIT
 
-  const getFields = async (bucket: any, measurement: string) => {
+  const getFields = async (
+    bucket: any,
+    measurement: string,
+    searchTerm?: string
```

Comment:
Why are we typing bucket as `any` type? When we have access to the Bucket type?

Reply:
Nice catch! Totally agree that we should use type `Bucket` instead of `any`. 

However, if I changed the type to `Bucket`, VS code would shows error message on line 79 and line 96
![Screen Shot 2022-06-02 at 4 45 08 PM](https://user-images.githubusercontent.com/14298407/171744424-592ed68a-9444-4f1e-9318-2f1a091072dd.png)

As you can see from the screenshot, there should be only 'user' and 'system' for `bucket.type` as [it is defined in openapi](https://github.com/influxdata/openapi/blob/3481defece3e8c70c47f21680857ac1a7c051a8b/src/common/schemas/Bucket.yml#L34-L40)

The reason why can line 79 check `bucket.type === 'sample'` is because the 'sample' buckets are hardcoded into the bucket list in the frontend [here](https://github.com/influxdata/ui/blob/master/src/shared/contexts/buckets.tsx#L152-L172)

A fix I can think of is to add a line `readonly type?: 'user' | 'system' | 'sample' `in [`Bucket` type definition](https://github.com/influxdata/ui/blob/master/src/types/buckets.ts#L5-L8) so that it's syntactically correct to use `Bucket`. 

What do you think? @wiedld

Reply:
I tired the fix I suggested above in [this commit](https://github.com/influxdata/ui/pull/4727/commits/309d391d55057b8051e93daf6a86e9a5be2276b9). Seems like lint on CircleCI is [complaining](https://app.circleci.com/pipelines/github/influxdata/ui/13546/workflows/732d3150-cd24-4a12-af40-7068932c2b2d/jobs/80137)

<img width="1199" alt="Screen Shot 2022-06-02 at 5 38 25 PM" src="https://user-images.githubusercontent.com/14298407/171750055-30eb1bdf-98f4-49f2-bdbe-7ebf0e14490b.png">

So I think using type `any` is fair unless you have other suggestions. I'm open to try!

Reply:
I think this is typescript doing it's job. And pointing out that the hardcoded sample buckets are not fitting the Bucket type. 😆 

That hardcoding is taking an existing bucket payload from the backend, and then hardcoding a subset of example buckets (without it being bucket CRUD objects) in the frontend, while adding an additional "type == sample" to the frontend. Then all our special case handling (of how we get this to work) looks for:
* the featureFlag `newQueryBuilder` being present.
* AND the bucket being this special Bucket object we created (which doesn't follow the type contract).

Ramifications of this decision:
* once we take this thing live, we have to update all the other conditional paths to handle non "type == sample" buckets. 
* we are not testing (as we dev) buckets which will exist in the real world. A.k.a. the other typed buckets -- such as system buckets, user buckets, etc.
* Is this what we want?

Alternatively, may I suggest:
* in this section: https://github.com/influxdata/ui/blob/master/src/shared/contexts/buckets.tsx#L152-L172
* If we have the featureFlag `newQueryBuilder` AND the sample buckets we seek are not present:
   * seed your sample buckets as `type: user` or `type: system`.
   * type the sample buckets per the expected API contract.
   * Then make sure all downstream code does not break.

Does this make sense?

Reply:
sample buckets are not api resources, and are not used as such for code generation which is where you see the branching off of the bucket type (cannot use a `from` but instead need to use a `sample`)

Reply:
Thanks @wiedld! I continue the suggestion from @drdelambre and fixed the complains from Typescript. Now `Bucket` type is working well!